app/src/main/java/com/intervaltimer/google/activities/MainActivity.java

```diff
@@ -142,51 +141,46 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
         }
 
         Intent receivedIntent = getIntent();
-        Log.i(LONG_INTENT_TAG, "Intent in onCreate() triggered");
+        Log.i(LOG_INTENT_TAG, "Intent in onCreate() triggered");
         handleIntent(receivedIntent);
     }
 
     @Override
     protected void onNewIntent(final Intent intent) {
         super.onNewIntent(intent);
-        Log.i(LONG_INTENT_TAG, "Intent triggered");
+        Log.i(LOG_INTENT_TAG, "Intent triggered");
         handleIntent(intent);
     }
 
     public void handleIntent(final Intent receivedIntent) {
         // EXERCISE_START
-        Toast show;
+
         if (receivedIntent != null) {
-//            show = Toast.makeText(getApplicationContext(),
-//                    "Intent: " + receivedIntent.toString(), Toast.LENGTH_LONG);
-            if (Objects.equals(receivedIntent.getAction(), Intent.ACTION_VIEW)) {
-                Log.i(LONG_INTENT_TAG, Objects.requireNonNull(receivedIntent.getData()).toString());
+            Uri data = receivedIntent.getData();
+            if (data != null && Objects.equals(receivedIntent.getAction(), Intent.ACTION_VIEW) &&
+                    Objects.equals(data.getPath(), "/start")) {
+
+                Log.i(LOG_INTENT_TAG, data.toString());
```

Comment:
What about a very short prefix to the log statement to give the reader of the log some context?